jgrapht-core/src/main/java/org/jgrapht/alg/util/UnionFind.java

```diff
@@ -132,6 +139,66 @@ public void union(T element1, T element2)
             parentMap.put(parent2, parent1);
             rankMap.put(parent1, rank1 + 1);
         }
+        count--;
+    }
+
+    /**
+     * Tests whether two elements are contained in the same set.
+     * @param element1 first element
+     * @param element2 second element
+     * @return true if element1 and element2 are contained in the same set, false otherwise.
+     */
+    public boolean connected(T element1, T element2){
+        return find(element1).equals(find(element2));
+    }
+
+    /**
+     * Returns the number of sets. Initially, all items are in their own set. The smallest number of sets equals one.
+     * @return the number of sets
+     */
+    public int count() {
```

Comment:
Consider changing the name to something like `numberOfSets()`, or something similar.